nvtabular/column_group.py

```diff
@@ -84,6 +87,34 @@ def __init__(
             columns = [_convert_col(col, tags=tags, properties=properties) for col in columns]
             self.columns: Columns = Columns(columns)
 
+    @staticmethod
```

Comment:
Reframing `ColumnGroup` as primarily a workflow graph node rather than a collection of columns makes me question whether the schema belongs here. It seems like this class already has more responsibilities than it should. I think we were talking yesterday about having `ColumnSchema` and `DatasetSchema` classes, so this might be the kind of thing we'd put in a `DatasetSchema` class.